framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1109,53 +1188,134 @@ int pcm_stop(struct pcm *pcm)
 		return -EINVAL;
 	}
 #ifdef CONFIG_AUDIO_MULTI_SESSION
-	if (ioctl(pcm->fd, AUDIOIOC_STOP, (unsigned long)pcm->session) < 0)
+	if (!pcm->running && ioctl(pcm->fd, AUDIOIOC_STOP, (unsigned long)pcm->session) < 0)
 #else
-	if (ioctl(pcm->fd, AUDIOIOC_STOP, 0) < 0)
+	if (!pcm->running && ioctl(pcm->fd, AUDIOIOC_STOP, 0) < 0)
 #endif
 		return oops(pcm, errno, "cannot stop channel");
```

Comment:
Which is correct? cannot stop channel or cannot stop player?

Reply:
I think this means cannot stop channel. Stopping player would be to call pcm_close()